### ISSUE TITLE
Site Recovery Azure to Azure replication: How to configure test failover network and target network when creating replicated VMs?

### ISSUE THREAD

**Issue by yilunpwc** [hashicorp/terraform-provider-azurerm#7039]
Hi,

I wonder if this feature is supported or not. I'm following https://www.terraform.io/docs/providers/azurerm/r/site_recovery_replicated_vm.html to create a replicated VM. It mostly worked fine, but is missing the test failover network and target network. Additionally, site_recovery_replicated_vm doesn't seem to expose arguments for these networks.

After creating the replicated VM in Terraform, if I go to Azure portal to manually configure the networks, it also displayed an error. So I think maybe some steps in Terraform's replication process did not finish correctly.

Failed to complete the operation.
Error: The properties of the virtual machine can't be configured. Ensure that the virtual machine is configured for protection successfully.

By the way, before creating the replicated VM, I also created a network mapping as described on https://www.terraform.io/docs/providers/azurerm/r/site_recovery_network_mapping.html. This step worked fine, and I can see the network mapping on portal. However I think this step is necessary for a successful VM replication, otherwise we won't be able to perform failover or failback. Maybe it's better to add it to site_recovery_replicated_vm's sample in the document?

Thanks for any suggestions.

**Comment by martenbohlin** [comment]
If pull request #5688 is approved it would be possible to set the target network. Once that is in place we can add support for test failover network as well.

**Comment by MitchDrage** [comment]
The above PR appears to have solved this requirement, however I'm finding that the static IP field isn't optional as described.
I'll raise a separate issue to track the static IP field.
![image](https://user-images.githubusercontent.com/25337396/95671327-50700d80-0be1-11eb-9d0d-cf2b10d50c52.png)

**Comment by mgpole** [comment]
Hi, I came across the same problem today with configuring ASR on a few VM's. Is adding support for test failover network still on the road map?

**Comment by favoretti** [comment]
@mgpole @MitchDrage Hi and thank you for logging the issue and providing feedback! Checking in here to see whether this actually has been addressed in the above PR or not?

**Comment by martenbohlin** [comment]
@favoretti No I implemented the target network. But I never got around to implement the **test** network.

**Comment by dsiperek-vendavo** [comment]
This is a blocker for us adopting ASR.

**Comment by katbyte** [comment]
It appears this issue is a duplicate of, or similar to, #19846 (Support for Test Failover Network and Test failover Settings for ASR), which has been marked as completed and resolved in v3.42.0, as such we are going to close this one out.

If this is still an issue on the latest version of the provider (v5.x) please do let us know with a comment or open a new issue referencing this one. Thank you!
